docs/getting-started.md

```diff
@@ -1,31 +1,392 @@
+
+    GetMetricByName(name types.NamespacedName, info CustomMetricInfo) (*custom_metrics.MetricValue, error)
+    GetMetricBySelector(namespace string, selector labels.Selector, info CustomMetricInfo) (*custom_metrics.MetricValueList, error)
+}
+```
+
+First, there's a method for listing all metrics available at any point in
+time.  It's used to populate the discovery information in the API, so that
+clients can know what metrics are available.  It's not allowed to fail (it
+doesn't return any error), and it should return quickly, so it's suggested
+that you update it asynchronously in real-world code.
+
+For this walkthrough, you can just return a few statically-named metrics,
+two that are namespaced, and one that's on namespaces themselves, and thus
+root-scoped:
+
+```go
+func (p *yourProvider) ListAllMetrics() []provider.CustomMetricInfo {
+    return []provider.CustomMetricInfo{
+        // these are mostly arbitrary examples
+        {
+            GroupResource: schema.GroupResource{Group: "", Resource: "pods"},
+            Metric:        "packets-per-second",
+            Namespaced:    true,
+        },
+        {
+            GroupResource: schema.GroupResource{Group: "", Resource: "services"},
+            Metric:        "connections-per-second",
+            Namespaced:    true,
+        },
+        {
+            GroupResource: schema.GroupResource{Group: "", Resource: "namespaces"},
+            Metric:        "work-queue-length",
+            Namespaced:    false,
+        },
+    }
+}
+```
+
+Next, you'll need to implement the methods that actually fetch the
+metrics. There are methods for fetching metrics describing arbitray Kubernetes
+resources, both root-scoped and namespaced-scoped.  Those metrics can
+either be fetched for a single object, or for a list of objects by
+selector.
+
+<details>
+
+<summary>Make sure you understand resources, scopes, and quantities before
+continuing.</summary>
+
+---
+
+#### Kinds, Resources, and Scopes
+
+When working with these APIs (and Kubernetes APIs in general), you'll
+often see references to resources, kinds, and scope.
+
+Kinds refer to types within the API.  For instance, `Deployment` is
+a kind.  When you fully-qualify a kind, you write it as
+a *group-version-kind*, or *GVK* for short.  The GVK for `Deployment` is
+`apps/v1.Deployment` (or `{Group: "apps", Version: "v1", Kind:
+"Deployment"}` in Go syntax).
+
+Resources refer to URLs in an API, in an abstract sense.  Each resource
+has a corresponding kind, but that kind isn't necessarily unique to that
+particular resource.  For instance, the resource `deployments` has kind
+`Deployment`, but so does the subresource `deployments/status`.  On the
+other hand, all the `*/scale` subresources use the kind `Scale` (for the
+most part).  When you fully-qualify a kind, you write it as
+a *group-version-resource*, or GVR for short.  The GVR `{Group: "apps",
+Version: "v1", Kind: "Deployment"}` corresponds to the URL form
```

Comment:
I was expecting this to look like: `{Group: "apps", Version: "v1", Resource: "Deployment"}` since it is a GVR...  but look like you request a resource by the GVK?

the statements:

``` When you fully-qualify a kind, you write it as
a *group-version-kind*, or *GVK* for short.
```

and 

```
 When you fully-qualify a kind, you write it as
a *group-version-resource*, or GVR for short.
```

seem to conflict

Reply:
lol, whoops, copy-paste error.  I'll fix that.